### PR TITLE
Sharding and query

### PR DESCRIPTION
**Pull request by richardschneider**
Fixes `sharding.query`.  All `interface-datastore` tests now pass.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -52,6 +52,7 @@
   },
   "contributors": [
     "David Dias <daviddias.p@gmail.com>",
-    "Friedel Ziegelmayer <dignifiedquire@gmail.com>"
+    "Friedel Ziegelmayer <dignifiedquire@gmail.com>",
+    "Richard Schneider <makaretu@gmail.com>"
```
Comment (dignifiedquire) [inline package.json]: no need to do this manually these lists are generated on release

File: src/sharding.js [hunk 2]
```diff
@@ -128,52 +128,45 @@ class ShardingDatastore {
...
     }
 
     if (q.filters != null) {
-      tq.filters = q.filters.map((f) => (e, cb) => {
+      let filters = q.filters.map((f) => (e, cb) => {
         f(Object.assign({}, e, {
           key: this._invertKey(e.key)
         }), cb)
       })
+      tq.filters.push(...filters)
```
[the remark this answers is not in the document]
Comment (richardschneider) [reply src/sharding.js]: okay, will downgrade

**Comment by richardschneider** [comment]
@dignifiedquire Could you look at the build?  Getting errors from `flow`; I have no knowledge of the tool.

Good news is that lint and all tests pass.

**Comment by richardschneider** [comment]
Now that we are building on windows, we need PR https://github.com/ipfs/interface-datastore/pull/12.

Any idea on when it will be approved and released to NPM?

File: src/sharding.js [hunk 3]
```diff
@@ -128,52 +127,45 @@ class ShardingDatastore {
   }
 
   query (q /* : Query<Buffer> */) /* : QueryResult<Buffer> */ {
-    const tq/* : Query<Buffer> */ = {
-      keysOnly: q.keysOnly
+    let tq/* : Query<Buffer> */ = {
```
Comment (dryajov) [inline src/sharding.js]: /let/const/

File: src/sharding.js [hunk 4]
```diff
@@ -128,52 +127,45 @@ class ShardingDatastore {
...
-      // TODO: transform
-      tq.prefix = q.prefix
+      tq.filters.push((e, cb) => {
+        cb(null, this._invertKey(e.key).toString().startsWith(q.prefix))
+      })
     }
 
     if (q.filters != null) {
-      tq.filters = q.filters.map((f) => (e, cb) => {
+      let filters = q.filters.map((f) => (e, cb) => {
```
Comment (dryajov) [inline src/sharding.js]: /let/const/

File: src/sharding.js (hunk 3 above)
Comment (richardschneider) [reply src/sharding.js]: I'm a bit old-school, I use const when the object won't change; which is the JS usage..

File: src/sharding.js (hunk 4 above)
Comment (richardschneider) [reply src/sharding.js]: got me, consistency is the hobglobin of little minds!

I'll make both changes

**Pull request merged**